src/DynamoCore/Scheduler/UpdateGraphAsyncTask.cs

```diff
@@ -59,6 +94,40 @@ internal bool Initialize(EngineController controller, WorkspaceModel workspace)
                 TargetedWorkspace = workspace;
 
                 ModifiedNodes = ComputeModifiedNodes(workspace);
+
+                // Detect cyclic dependencies in graph
+                var cyclicNodes = new List<NodeModel>();
+
+                var visited = new HashSet<NodeModel>();
+                var recursed = new HashSet<NodeModel>();
+                foreach (var node in ModifiedNodes)
+                {
+                    if(!IsGraphCyclic(node, visited, recursed, cyclicNodes))
+                    {
+                        var warnings = controller.LiveRunnerCore.BuildStatus.Warnings.ToList();
+                        foreach(var warning in warnings)
+                        {
+                            if(warning.ID == WarningID.InvalidStaticCyclicDependency && 
+                                node.GUID == warning.GraphNodeGuid)
+                            {
+                                node.ClearErrorsAndWarnings();
```

Comment:
So these warnings are from a previous execution when the graph was in a cyclic state?

Reply:
Yes, if a cyclic warning is present for this node from a previous execution and the graph is found to be acyclic for the node in the current execution, the warning will be cleared for that node.

Reply:
`ClearErrorsAndWarnings` will clear everything (all previous persistent warnings or errors)
Do you think there might be a case where older (persistent) errors should not be cleared ?

Reply:
Yes, it will clear everything but I'm reducing the surface area of it clearing warnings only when the node in question had a cyclic warning before, but you're right, this will still clear any and all other warnings that the node previously had as well. However, I can't think of a scenario where that would do any harm as I'm relying on the fact that the same warnings should reappear if the node is re-executed.

Reply:
Most of the internal (not created by API users) are indeed mostly regenerated when re executing. We do have example of non execution related warnings (like the Gizmo warning)
I guess the issue would be if API users create warnings that do not get recreated by re execution
Not a big issue though...warning infrastructure does not support it yet